src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -1,69 +1,621 @@
+            {
+                return;
+            }
+
+            string matchingKeyword;
+            if (!IsKeyword(symbol.Name, out matchingKeyword))
+            {
+                return;
+            }
+
+            // IsAbstract returns true for both abstract class members and interface members.
+            if (symbol.IsVirtual || symbol.IsAbstract)
+            {
+                context.ReportDiagnostic(
+                    symbol.CreateDiagnostic(
+                        MemberRule,
+                        FormatSymbolName(symbol),
+                        matchingKeyword));
+            }
+        }
+
+        private void AnalyzeMemberParameterRule(SymbolAnalysisContext context)
+        {
+            var method = (IMethodSymbol)context.Symbol;
+            if (method.GetResultantVisibility() != SymbolVisibility.Public)
+            {
+                return;
+            }
+
+            // IsAbstract returns true for both abstract class members and interface members.
+            if (!method.IsVirtual && !method.IsAbstract)
+            {
+                return;
+            }
+
+            foreach (var parameter in method.Parameters)
+            {
+                string matchingKeyword;
+                if (IsKeyword(parameter.Name, out matchingKeyword))
+                {
+                    context.ReportDiagnostic(
+                        parameter.CreateDiagnostic(
+                            MemberParameterRule,
+                            FormatSymbolName(method),
+                            parameter.Name,
+                            matchingKeyword));
+                }
+            }
+        }
+
+        private bool IsKeyword(string name, out string keyword)
+        {
+            if (s_caseSensitiveKeywords.TryGetValue(name, out keyword))
+            {
+                return true;
+            }
+
+            return s_caseInsensitiveKeywords.TryGetKey(name, out keyword);
+        }
+
+        // Format the symbol name in a way consistent with FxCop's display for this rule.
+        private static string FormatSymbolName(ISymbol symbol)
+        {
+            return symbol.ToDisplayString(
+                // This format omits the namespace.
+                SymbolDisplayFormat.CSharpShortErrorMessageFormat
+                    // Turn off the EscapeKeywordIdentifiers flag (which is on by default), so that
+                    // a method named "@for" is displayed as "for".
+                    // Turn on the UseSpecialTypes flat (which is off by default), so that parameter
+                    // names of "special" types such as Int32 are displayed as their language alias,
+                    // such as int for C# and Integer for VB.
+                    .WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.UseSpecialTypes));
         }
+
+        private ImmutableHashSet<string> s_caseSensitiveKeywords = new string[]
```

Comment:
readonly?

You should either make this static or rename the field to avoid the `s_` prefix, which should only be used for static fields. Also consider moving the fields to the beginning of the file.